src/main/java/com/internet/shop/controller/user/GetUserDetailsController.java

```diff
@@ -18,7 +18,12 @@
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
-        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        Long userId;
+        try {
+            userId = Long.parseLong(req.getParameter(USER_ID));
+        } catch (NumberFormatException exception) {
+            userId = (Long) req.getSession().getAttribute(USER_ID);
+        }
```

Comment:
same question: what is the case when we don't receive userId as a parameter?

Reply:
Agree
No need for this